src/axom/sidre/core/View.cpp

```diff
@@ -295,6 +295,50 @@ View* View::reallocate(const DataType& dtype)
   return this;
 }
 
+/*
+ *************************************************************************
+ *
+ * Reshape an array View.
+ *
+ *************************************************************************
+ */
+View* View::reshapeArray(int ndims, const IndexType* shape)
+{
+  if(m_state != BUFFER && m_state != EXTERNAL)
+  {
+    SLIC_WARNING(SIDRE_VIEW_LOG_PREPEND
+                 << "View can only reshape array states (BUFFER or EXTERNAL.");
+    return this;
+  }
+
+  IndexType oldSize = 1;
+  for(const auto& s : m_shape)
+  {
+    oldSize *= s;
+  }
```

Comment:
Suggestion (clarity): 
This might be clearer is `oldSize` was simply set to `this->getNumElements()`

Reply:
Agreed.